.pre-commit-config.yaml

```diff
@@ -75,3 +75,31 @@ repos:
           - flake8-print
         args:
           - --per-file-ignores=tests/*.py:T201 # prints are allowed in test files
+    # Use mypy for static type checking.
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: 'v0.991'
+    hooks:
+      - id: mypy
+        files: >
+          (?x)^(
+            src/snowflake/snowpark/modin/pandas/frontend/indexing.py|
+            src/snowflake/snowpark/modin/pandas/frontend/snow_dataframe.py|
+            src/snowflake/snowpark/modin/pandas/frontend/snow_series.py|
+            src/snowflake/snowpark/modin/pandas/translation/_internal/.*\.py|
+            src/snowflake/snowpark/modin/pandas/frontend/snow_partition_iterator.py|
+            src/snowflake/snowpark/modin/pandas/translation/compiler/snowflake_query_compiler.py|
+            src/snowflake/snowpark/modin/pandas/translation/default2pandas/.*\.py|
+            src/snowflake/snowpark/modin/pandas/utils/.*\.py|
+          )$
```

Comment:
Are these being excluded? If so, can we remove the exclusions?

Reply:
These should not be excluded - these should be the only files on which mypy runs
"(?x) enables the VERBOSE flag for the subsequent regular expression, which [ignores most whitespace and supports comments](https://docs.python.org/3/library/re.html#re.VERBOSE)" from https://mypy.readthedocs.io/en/stable/config_file.html#confval-exclude